packages/flame/lib/src/widgets/sprite_button.dart

```diff
@@ -54,12 +74,17 @@ class SpriteButton extends StatelessWidget {
     this.errorBuilder,
     this.loadingBuilder,
     Key? key,
-  })  : _buttonsFuture = (() => Future.wait([
-              Future.value(sprite),
-              Future.value(pressedSprite),
-            ])),
+  })  : _buttonsFuture = Future.wait([
+          sprite,
+          pressedSprite,
+        ]),
         super(key: key);
 
+  /// loads image from asset [path], [pressedPath] and renders it as Widget.
+  /// it has [loadingBuilder] while the image is loading.
+  /// To render without loading or gapless playback when the [path] change,
+  /// consider load an image from somewhere and directly pass it to
+  /// default constructor
```

Comment:
```suggestion 
  /// Loads the images from the asset [path] and [pressedPath] and renders 
  /// it as a widget.
  ///
  /// It will use the [loadingBuilder] while the image from [path] is loading.
  /// To render without loading, or when you want to have a gapless playback 
  /// when the [path] value changes, consider loading the image beforehand 
  /// and direct pass it to the default constructor.
```